configure.ac

```diff
@@ -1421,6 +1427,36 @@ if test "x$use_external_signer" != xno; then
 fi
 AM_CONDITIONAL([ENABLE_EXTERNAL_SIGNER], [test "x$use_external_signer" = "xyes"])
 
+dnl Do not compile with syscall sandbox support when compiling under the sanitizers.
+dnl The sanitizers introduce use of syscalls that are not typically used in bitcoind
+dnl (such as execve when the sanitizers execute llvm-symbolizer).
+if test x$use_sanitizers != x; then
+    AC_MSG_WARN(Specifying --with-sanitizers forces --without-seccomp since the sanitizers introduce use of syscalls not allowed by the bitcoind syscall sandbox (-sandbox=<mode>).)
+   seccomp_found=no
+fi
```

Comment:
I'm planning on tackling sanitizer compatibility in a follow-up PR. It is not entirely trivial since the different sanitizers introduce the use of different sets of syscalls, but it should be doable. However, I prefer keeping that outside of this PR to keep things focused on the "standard" non-sanitizer case first :)